### DISCUSSION TITLE
Installer doesn't do anything

### DISCUSSION THREAD

**Discussion by ravener** [ppy/osu#25188] [Q&A] [answered]
### Type

Crash to desktop

### Bug description

I opened install.exe and i get the osu! cookie with a loading bar, after it's done it's closed and nothing happens, but apparently osu! is shown on my Windows' list of programs but no desktop/start menu shortcut, so I tried to go into the directory and manually run the exe and see what happens and this is what I get
```
Ravener@Aster MINGW64 ~/AppData/Local/osulazer
$ ./osu\!.exe
Unhandled exception. System.BadImageFormatException: Could not load file or assembly 'System.Text.RegularExpressions, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The module was expected to contain an assembly manifest.
File name: 'System.Text.RegularExpressions, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
   at Squirrel.SemanticVersion.TryParse(String version, SemanticVersion& value)
   at Squirrel.SemanticVersion.Parse(String version) in ./NuGet/SemanticVersion.cs:line 151
   at Squirrel.SemanticVersion..ctor(String version) in ./NuGet/SemanticVersion.cs:line 22
   at Squirrel.UpdateManager.CurrentlyInstalledVersion(String executable) in ./UpdateManager.cs:line 214
   at Squirrel.SquirrelAwareApp.HandleEvents(SquirrelHook onInitialInstall, SquirrelHook onAppUpdate, SquirrelHook onAppObsoleted, SquirrelHook onAppUninstall, SquirrelRunHook onEveryRun, String[] arguments) in ./SquirrelAwareApp.cs:line 96
   at osu.Desktop.Program.setupSquirrel()
   at osu.Desktop.Program.Main(String[] args)

```
My first idea after seeing this was to see if updating .NET fixes it so I went and installed the latest .NET 6.0.23 Desktop Runtime from the .NET website, but that didn't change anything. (apparently looking at the error from windows event viewer it reports .NET 6.0.22 still so my installation didn't take effect it seems, or is the runtime version declared by osu! and just has to match?)

Any ideas what else I can troubleshoot?

### Version

2023.1008.1

### Logs

N/A, The game doesn't start to the point of emitting logs, the installer just extracted it and it still feels like it's not actually installed? There is still an unextracted .nupkg file in there.

**Comment by longnguyen2004** [comment]
Can you run outside of MSYS2 shell? (i.e. cmd/PowerShell)

**Comment by ravener** [comment]
> Can you run outside of MSYS2 shell? (i.e. cmd/PowerShell)

For some reason cmd.exe/Powershell both do not show any error but it still doesn't launch so they somehow hide the error when the same thing is probably happening. The issue is not with the cmd/powershell/bash issue though since the error is clearly visible in the Windows Event Viewer even when just double clicking from Explorer

**Comment by bdach** [comment]
I have no idea why this is happening but this must be pretty clearly something off with your system. I have never seen this before, there are zero upstream reports of this.

What are you running this on? Maybe make sure you installed the correct bitness binaries or something.

**Reply by ravener** [reply]
@bdach I'm running this on Windows 10 Pro Version 22H2 (OS Build 19045.2965) (64-bits), I'm on a Dell Latitude 7390 that I literally bought a week ago and it's the preinstalled OS that came with it so it's pretty clean. It sounds like something is wrong the .NET's runtime files? Any ideas to try, like if there's a way to refresh the .NET stuff I have, since installing a newer runtime didn't get picked up apparently. (It still runs on .NET 6.0.22 when I tried installing 6.0.23)

**Reply by bdach** [reply]
Can you show (with a screenshot) which version are you installing specifically?

**Reply by ravener** [reply]
if you mean the .NET it's the 'Desktop Runtime' from https://dotnet.microsoft.com/en-us/download/dotnet/6.0 which did not help at all because osu! is still being run with the older .NET that came with my system (6.0.22)

If you mean osu! it's just the latest install.exe from github releases.

I'm not sure what's there to screenshot, sorry.

**Reply by bdach** [reply]
Which of the windows binaries are you downloading? x64, x86 or arm64?

**Reply by ravener** [reply]
for .NET, x64 obviously, for osu! there is only one binary.

**Reply by bdach** [reply]
Then I'm really not sure sorry. That is most unusual.

I'd maybe try fully uninstalling and then re-installing .NET, if that is something that you can do. Failing that, I'm not sure where to even begin.

**Reply by ravener** [reply]
Yeah no worries, I figured this was an unusual thing, I'll try a few things by myself and see if something improves.

**Reply by ravener** [reply, marked as answer]
Good news, the problem was simply a corrupted download of osu!, redownloading the file installed successfully, apologize for the inconvenience, I simply did not anticipate a corrupted download can cause issues like this.

**Reply by peppy** [reply]
As a note, we bundle the .net runtime. Downloading it is not required and will not be used even if you do.
